### DISCUSSION TITLE
How to unit test a RavenDB

### DISCUSSION THREAD

**Discussion by giscus** [linkdotnet/Blog.Discussions#9] [General]
# How to unit test a RavenDB

RavenDB is a well known open-source document-oriented databse for .NET. And of course we want to test our logic and not only locally while developing, but also our continuous integration pipeline should be able to run our tests. So let's tackle exactly that.

https://steven-giesel.com/blogPost/0cbe9770-40e7-43b1-980f-d1d3a8b5203d

**Comment by tobiaszuercher** [comment]
i strongly believe that this repository pattern doesn't make sense when using ravendb.

there is a good talk about why you shouldn't abstract ravendb: https://www.youtube.com/watch?v=BruPcBDoXZw

**Reply by linkdotnet** [reply]
@tobiaszuercher I do see your point. Especially because RavenDB does not "suffer" the same issues as lets say EF Core where a `DbContext` is not mockable (to increase testability). 

Maybe I could have taken another example to demonstrate the situation better. I hope still the point I wanted to demonstrate could be transferred :D
